pypit/bpmimage.py

```diff
@@ -109,61 +90,22 @@ def build(self):
         self.bpm
 
         """
-        if self.reduce_badpix == 'bias':
-            # Get all of the bias frames for this science frame
-            if self.msbias is None:
-                msgs.warn("No bias frame provided!")
-                msgs.info("Not preparing a bad pixel mask")
-                return False
-            # TODO -- Deal better with this datasec kludge
-            datasec = []
-            for i in range(self.settings['detector']['numamplifiers']):
-                sdatasec = "datasec{0:02d}".format(i+1)
-                datasec.append(self.settings['detector'][sdatasec])
-            # Construct
-            self.bpm = arprocimg.badpix(self.msbias, self.settings['detector']['numamplifiers'], datasec)
-        else:
-            self.bpm = self.spectro_class.bpm(binning=self.binning, det=self.det,
-                                              shape=self.shape)
-        # Return
-        return self.bpm
+        # Will raise an exception if could not construct the BPM
+        self.bpm_img = self.spectrograph.bpm(shape=self.shape, filename=self.filename,
+                                             det=self.det) if self.msbias is None else \
+                            arprocimg.badpix(self.msbias,
+                                    self.spectrograph.detector[self.det-1]['numamplifiers'],
```

Comment:
While it may be a bit risky, I should love to have a 'thisdet' dict or something like that, e.g.

```
self.spectrograph.thisdet = self.spectrograph.detector[self.det-1]
```

Of course, this makes `spectrograph` mutable..

But I've grown tired of passing in self.det-1 (and needing to pass around `det`
just to do this).  Let's discuss.

Reply:
I see no problem getting an instance of this, for example, at the top of the routine use: ```thisdet = self.spectrograph.detector[self.det-1]``` and then call things like ```thisdet['numamplifiers']```. It would still therefore be immutable. I don't see the advantage of including this into the spectrograph class (or, for that matter, including it in this bpm class)

Reply:
Fair point.